src/uu/tail/src/tail.rs

```diff
@@ -567,742 +98,179 @@ fn uu_tail(mut settings: Settings) -> UResult<()> {
         the input file is not a FIFO, pipe, or regular file, it is unspecified whether or
         not the -f option shall be ignored.
         */
-        if files.no_files_remaining(&settings) {
-            if !files.only_stdin_remaining() {
-                show_error!("{}", text::NO_FILES_REMAINING);
-            }
-        } else if !(settings.stdin_is_pipe_or_fifo && settings.paths.len() == 1) {
-            follow(files, &settings, watcher_rx, orphans)?;
+        // TODO: can I do this check in watch::follow?
```

Comment:
There are a lot of TODO notes in here, which is fine, but because this is an open project, they need some context for everyone else who is reading the code.

Reply:
Sorry for that. I won't leave any TODOs when I'm finished, but this is just an intermediate pr, so I would like to keep them for now.

Reply:
Ok I found a solution, so we can both be happy. I removed all TODOs in tail, so they don't end up in the main branch. I'll revert that commit later. No big deal.

Reply:
It looks like you removed **all** TODOs, even those that were there before this refactor. This is very inconvenient.

Reply:
Yeah I wasn't suggesting to remove them, I just wanted them to be a bit more verbose so other people might pick them up more easily if they come across them in the code :)